File: nodejs/src/api/handlers/results/createResult/createResultHandler.js
'use strict';

const AWS = require('aws-sdk')
const db = new AWS.DynamoDB.DocumentClient({})

const { POLLS_TABLE_NAME } = process.env

const response = (statusCode, message) => {
    return {
        statusCode: statusCode,
        body: JSON.stringify(message)
    }
}

const createResult = async (event) => {
  try {
  
    const result = JSON.parse(event.body)

    console.log(JSON.stringify(result))

    let params = {
      TableName: POLLS_TABLE_NAME,
      Item: result
    }

    let res = await db.put(params).promise()

    if (res) {
        return response(200, result)
    } else {
        return response(500, 'Error at creating the result')
    }
  } catch (err) {
    console.error('err: ', err)
    return response(err.statusCode, err)
  }
}

module.exports = {
  createResult
}

File: python/src/api/handlers/results/createResult/createResultHandler.py
try:
    import unzip_requirements
except ImportError:
    pass

import boto3
import os
import simplejson as json

dynamodb_client = boto3.resource('dynamodb')
TABLE_NAME = os.environ['FORMS_TABLE_NAME']

def create_response(code, message):
  return {
    'statusCode': code,
    'headers': {'Content-Type': 'application/json'},
    'body': json.dumps(message)
  }

def createResult(event, context):

    result = json.loads(event['body'])

    params = {
      'Item': result
    }

    print('params', params)

    try:
      table = dynamodb_client.Table(TABLE_NAME)
      response = table.put_item(**params)
    except Exception as ex:
      print('Ex: ', ex)
      return create_response(500, 'Error creating the Result')

    print('Response', response)

    if response['ResponseMetadata']['HTTPStatusCode'] != 200:
      return create_response(500, 'Error creating the Result')

    return create_response(200, params['Item'])

